### DISCUSSION TITLE
Support target: pseudoclass

### DISCUSSION THREAD

**Discussion by ctheune** [tailwindlabs/tailwindcss#2056] [Ideas]
Hi,

thanks for a fantastic product - I'm having a great time using it and am also looking forward to the release of the components!

Anyway, I stumbled over a pseudoclass that isn't supported (yet): ":target" and guess you might want to consider it for inclusion. Not sure whether I can contribute the PR, but if it's generally welcome I can try and figure out what to do, I guess it's a small change, but a CSS magician might have a different opinion. :)

Cheers!

**Comment by philippbosch** [comment]
Adding more pseudo classes can increase CSS file size (at least in development) and build times significantly. I guess that's why a couple of them (e.g. `:empty`) are intentionally excluded from Tailwind's default configuration. But you can easily add support for `:target` to your project by adding this to your `tailwind.config.js`:

```js
module.exports = {
  // ...
  plugins: [
    function({ addVariant, e }) {
      addVariant('target', ({ modifySelectors, separator }) => {
        modifySelectors(({ className }) => {
          return `.${e(`target${separator}${className}`)}:target`;
        });
      });
    },
  ]
};
```

**Reply by ctheune** [reply]
Right - I did find this in the docs and thanks for the specific snippet. My assumption was that the missing target was an oversight so maybe for future reference you might want to add the reasoning you gave me to the docs?

Thanks again and stay safe!

**Reply by ctheune** [reply]
Actually I need to pick this up again. I added that code (and compared it to the docs) and even added the `const plugin = require('tailwindcss/plugin')` part to my tailwind.config.js. However, I'm not getting any 'target' generated into my final CSS. I know that the other parts of my config do apply because custom colors, fonts, etc. do show up properly... 

Unfortunately I'm getting the silent treatment when building, so I'm a bit lost on how to debug this ... 

```
$ npm run build

> tailwind@1.0.0 build /Users/ctheune/Code/fc.directory/tailwind
> postcss customerui/src/tailwind.css -o customerui/customerui.css; cp customerui/customerui.css ../src/fc/directory/customerui/resources/css/customerui.css
$ grep "target" customerui/customerui.css
$

```

**Reply by philippbosch** [reply]
@ctheune My guess would be that you did not configure any `target:` variant classes to be generated. Please have a look at [this part of the docs](https://tailwindcss.com/docs/configuration/#variants).

If you want something like `target:bg-pink-500` to be generated, you need something like this in your _tailwind.config.js_:

```js
  variants: {
    backgroundColor: ['responsive', 'hover', 'target'],
  },
```

**Reply by bsor-dev** [reply]
> @ctheune My guess would be that you did not configure any `target:` variant classes to be generated. Please have a look at [this part of the docs](https://tailwindcss.com/docs/configuration/#variants).
> 
> If you want something like `target:bg-pink-500` to be generated, you need something like this in your _tailwind.config.js_:
> 
> ```js
>   variants: {
>     backgroundColor: ['responsive', 'hover', 'target'],
>   },
> ```

I got this error after I added `"target"` in tailwind.config.js

```
Error: Your config mentions the "target" variant, but "target" doesn't appear to be a variant. Did you forget or misconfigure a plugin that supplies that variant?
```

I'm using v2.0.0. Am I outdated?

**Comment by ctheune** [comment]
Ah, alright. Thanks. I did see that but didn't make the connection between those two mechanisms properly. Sorry for that.

**Reply by philippbosch** [reply]
All good. Happy to help.
